File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/ride/SaveRideUseCase.java
package com.lattis.ellipse.domain.interactor.ride;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.model.Ride;
import com.lattis.ellipse.domain.repository.RideRepository;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * Created by ssd3 on 4/13/17.
 */

public class SaveRideUseCase extends UseCase<Ride>

{

    private RideRepository rideRepository;
    private Ride ride;

    @Inject
    public SaveRideUseCase(ThreadExecutor threadExecutor,
                           PostExecutionThread postExecutionThread,
                           RideRepository rideRepository) {
        super(threadExecutor, postExecutionThread);
        this.rideRepository = rideRepository;
    }

    public SaveRideUseCase withRide(Ride ride) {
        this.ride = ride;
        return this;
    }


    @Override
    protected Observable<Ride> buildUseCaseObservable() {
        return this.rideRepository.saveRide(ride);
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/ride/SaveRideUseCase.kt
package com.lattis.domain.usecase.ride

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.RideRepository
import com.lattis.domain.usecase.base.UseCase
import com.lattis.domain.models.Ride

import javax.inject.Inject

import io.reactivex.rxjava3.core.Observable

class SaveRideUseCase @Inject
constructor(threadExecutor: ThreadExecutor,
            postExecutionThread: PostExecutionThread,
            private val rideRepository: RideRepository
) : UseCase<Ride>(threadExecutor, postExecutionThread) {
    private lateinit var ride: Ride

    fun withRide(ride: Ride): SaveRideUseCase {
        this.ride = ride
        return this
    }

    override fun buildUseCaseObservable(): Observable<Ride> {
        return this.rideRepository.saveRide(ride)
    }
}
